### PR TITLE
Alter react lifecycle methods to match current deprecation

### PR DESCRIPTION
**Pull request by mikehardy**
This is the result of running the approved tool for the purpose:
`npx react-codemod rename-unsafe-lifecycles` answering '.' and 'javascript with flow'

This is connected to improvement of the out-of-the-box development experience
in ReactXP TodoList sample which still uses this navigation library

https://github.com/microsoft/reactxp/pull/1214

Without this commit merged, the native app throws warnings immediately,
with it the dev experience should be completely warning free

### CODE REVIEW

**Comment by mikehardy** [comment]
@dryganets - you appear to be El Jefe around here, what do you think?

**Comment by dryganets** [comment]
@mikehardy, the change looks fine to me.
You probably want to keep both safe and unsafe versions to make sure that it will work with both versions of react.

It would be good to get in touch with some folks who is working on Skype.
I'm not sure if they upgraded to the react with unsafe methods yet and if those are connected through resub.
@rigdern could help you to connect with right people or correct me if I'm wrong.

I left Skype a year ago, so I likely to have an outdated context.

**Comment by mikehardy** [comment]
Resub has the unsafe methods now - it was a big change https://github.com/microsoft/ReSub/pull/118 - but done late October 2019 - no idea on Skype status and I have no contacts myself, sorry. I just offer patches to ReactXP mostly, as a motivated consumer of the tech

**Review by dryganets — APPROVED** [review]

**Comment by deregtd** [comment]
I'm not sure who still has admin access to this repo.  @erictraut ?

**Comment by erictraut** [comment]
I don't have admin access to this repo. I think @dryganets set it up, so I'm guessing he has access. Sergei, if that's correct, could you please switch admin access over to me?

**Comment by dryganets** [comment]
@erictraut, @deregtd this is Microsoft repository.
So, we need someone inside the company to take this change in.
Originally I gave the write permissions to everyone in the Davids team, it should be available through the Microsoft opensource tool.

let me check if I still can merge it :)
Technically I shouldn't have a permission anymore.

**Comment by dryganets** [comment]
nope, no write access.

**Comment by dryganets** [comment]
@rigdern has committed to the https://github.com/Microsoft/v8-docker-buildscript recently. That repository is in the same permissions group.

**Comment by rigdern** [comment]
I just gave @erictraut permissions to the repo.

Also, if we need a Skype contact, @danmuresan is a current Skype engineer with access to Skype's open source repos.

**Review by danmuresan — APPROVED** [review]

**Comment by danmuresan** [comment]
Approved. We will try to add more Skype engineers to the permission group in the near future

**Comment by mikehardy** [comment]
Great! Thanks everyone

**Comment by deregtd** [comment]
I think you need to merge it too, Dan. :)

**Comment by mikehardy** [comment]
@danmuresan I think it's ready for merge and just needs the magical button press ;-)

**Comment by danmuresan** [comment]
Sorry about the delay, I thought the requester can merge the PR once approved. Done

**Pull request merged**
